src/gui/categoryfilterwidget.cpp

```diff
@@ -168,7 +168,12 @@ void CategoryFilterWidget::callUpdateGeometry()
 
 QSize CategoryFilterWidget::sizeHint() const
 {
-    return viewportSizeHint();
+    QSize size;
+    // Height should be exactly the height of the content
+    size.setHeight(sizeHintForRow(0) * (model()->rowCount() + 0.5));
+    // Width should be exactly the width of the content
+    size.setWidth(sizeHintForColumn(0));
+    return size;
```

Comment:
IMO, it would be better to change it the following way:
```c++
return {
    // Width should be exactly the width of the content
    sizeHintForColumn(0)
    // Height should be exactly the height of the content
    , sizeHintForRow(0) * (model()->rowCount() + 0.5)};
```